### PR TITLE
update package.jsons, CCI, nvmrc, etc

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
Upgrades LMS from Node 14 to Node 16 

## WHY
- Node 14 is super old. For example, it's preventing us from upgrading to Vite 5.x 
- To minimize entropy, I'm upgrade Node two versions per deploy, rather than 4 versions at once.

## HOW
- Fix a few functions/specs that were hiding errors within promise rejections (see card for more details)
- update CCI and NVMRCs
- Configure CircleCI to use compute resources for type `xlarge` for `lms_node_build`, to avoid timeouts caused by longer jest execution times. We should be able to revert this config change on or after Node 20, per this issue: https://github.com/jestjs/jest/issues/11956

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Product-Board-30f97e4eb01246dbb8264253fb385073?p=b54cf585ebff4e3489a98c892ab42ce3&pm=s

### What have you done to QA this feature?
- [x] As a staffer, click into a Connect title card, ensure no rendering errors
- [x] Login with Gmail
- [x] As a staffer, complete an Evidence question
- [x] As a student, complete a Connect activity
- [x] Login as teacher, assign an activity
- [x] monitor Sidekiq for unusual errors 
- [x] In Heroku build logs, ensure Heroku is using Node 16

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? |

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Diagnostic/actions/titleCards.ts [hunk 1]
```diff
@@ -12,10 +12,11 @@ function startListeningToTitleCards() {
 function loadTitleCards(): (any) => void {
   return (dispatch) => {
     TitleCardApi.getAll(DIAGNOSTIC_TITLE_CARD_TYPE).then((body) => {
-      const titleCards = body.title_cards.reduce((obj, item) => {
+      let titleCards = body.title_cards || []
```
Comment (happythenewsad) [inline services/QuillLMS/client/app/bundles/Diagnostic/actions/titleCards.ts:15]: The test uses an empty mock for `body`. Thus, `body.<undefined>.reduce()` yields a TypeError in Node 16

File: services/QuillLMS/client/app/bundles/Diagnostic/test/actions/questions.test.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
   IncorrectSequenceApi: mockIncorrectSequenceApi,
   QuestionApi: mockQuestionApi,
 }))
 
-import { mockDispatch as dispatch, } from '../__mocks__/dispatch'
-
-import { SENTENCE_COMBINING_TYPE } from '../../libs/questions_api'
-
-import questionActions from '../../actions/questions'
+window.alert = jest.fn();
```
Comment (samanthamjohn) [inline services/QuillLMS/client/app/bundles/Diagnostic/test/actions/questions.test.ts]: this looks suspicious--but it seems you use `jest.fn()` later in the test. I'd move it closer to the call site to reduce confusion.

**Review by samanthamjohn — APPROVED** [review]
I think it looks good, there was one test thing that looked a little off, but if you have a good reason for it, then I think it's good to go. 
Thanks for doing this!

File: services/QuillLMS/client/app/bundles/Diagnostic/actions/titleCards.ts (hunk 1 above)
Comment (anathomical) [reply services/QuillLMS/client/app/bundles/Diagnostic/actions/titleCards.ts:15]: Take or leave this, but could we get away with safe navigation?  `const titleCards = body?.title_cards?.reduce(...) || {}`  I don't know if that's actually better, but it is a pattern we use fairly often in Ruby.

**Review by anathomical — APPROVED** [review]
One thought on handling null values, but this looks fine to go as-is to me.

File: services/QuillLMS/client/app/bundles/Connect/test/actions/questions.test.ts [hunk 3]
```diff
@@ -19,6 +19,8 @@ import { SENTENCE_COMBINING_TYPE } from '../../libs/questions_api';
 
 import questionActions from '../../actions/questions';
 
+window.alert = jest.fn();
```
Comment (dandrabik) [inline services/QuillLMS/client/app/bundles/Connect/test/actions/questions.test.ts]: Is this leftover from debugging?

File: services/QuillLMS/client/app/bundles/Diagnostic/test/actions/questions.test.ts (hunk 2 above)
Comment (happythenewsad) [reply services/QuillLMS/client/app/bundles/Diagnostic/test/actions/questions.test.ts:14]: My thinking was that this line is functioning as a file-wide mock, and we place these types of mocks directly below `import` statements.

**Review by dandrabik — APPROVED** [review]
Thanks for making the QA test plan very clear. Looks good if your QA looks good (one small comment/question). Probably worth alerting support that you are making this change when deploying, since it affects the whole site.

**Review by emilia-friedberg — APPROVED** [review]
LGTM. Excited for this upgrade!

File: services/QuillLMS/client/app/bundles/Connect/test/actions/questions.test.ts (hunk 3 above)
Comment (happythenewsad) [reply services/QuillLMS/client/app/bundles/Connect/test/actions/questions.test.ts]: No longer needed; thanks

**Pull request merged**